src/types.rs

```diff
@@ -98,13 +98,15 @@ impl FeeRate {
 
     /// Calculate fee rate from `fee` and weight units (`wu`).
     pub fn from_wu(fee: u64, wu: usize) -> FeeRate {
-        Self::from_vb(fee, wu.vbytes())
+        let vbytes = wu as f32 / 4.0;
+        let rate = fee as f32 / vbytes;
+        Self::new_checked(rate)
```

Comment:
With your change to the wu -> vb conversion, `Self::from_vb(fee, wu.vbytes())` does exactly the same thing as the lines above.